src/node/history.h

```diff
@@ -858,105 +784,9 @@ namespace ccf
       return consensus->on_request({id, request, caller_cert, frame_format});
     }
 
-    struct PendingInsert
-    {
-      PendingInsert(
-        kv::TxHistory::RequestID i,
-        kv::Version v,
-        std::shared_ptr<std::vector<uint8_t>> r) :
-        id(i),
-        version(v),
-        replicated(std::move(r)),
-        next(nullptr),
-        prev(nullptr)
-      {}
-
-      kv::TxHistory::RequestID id;
-      kv::Version version;
-      std::shared_ptr<std::vector<uint8_t>> replicated;
-      PendingInsert* next;
-      PendingInsert* prev;
-    };
-
-    SpinLock version_lock;
-    snmalloc::DLList<PendingInsert, std::nullptr_t, true> pending_inserts;
-
-    void add_pending(
-      kv::TxHistory::RequestID id,
-      kv::Version version,
-      std::shared_ptr<std::vector<uint8_t>> replicated) override
-    {
-      add_result(id, version, replicated->data(), replicated->size());
-    }
-
-    void flush_pending() override
-    {
-      snmalloc::DLList<PendingInsert, std::nullptr_t, true> pi;
-      {
-        std::lock_guard<SpinLock> vguard(version_lock);
-        pi = std::move(pending_inserts);
-      }
-
-      PendingInsert* p = pi.get_head();
-      while (p != nullptr)
-      {
-        add_result(p->id, p->version, *p->replicated);
-        p = p->next;
-      }
-    }
-
-    void add_result(
-      kv::TxHistory::RequestID id,
-      kv::Version version,
-      const std::vector<uint8_t>& replicated) override
-    {
-      add_result(id, version, replicated.data(), replicated.size());
-    }
-
-    void add_result(
-      RequestID id,
-      kv::Version version,
-      const uint8_t* replicated,
-      size_t replicated_size) override
-    {
-      append(replicated, replicated_size);
-
-      auto consensus = store.get_consensus();
-
-      if (consensus != nullptr && consensus->type() == ConsensusType::BFT)
-      {
-        if (on_result.has_value())
-        {
-          auto root = get_replicated_state_root();
-          LOG_DEBUG_FMT("HISTORY: add_result {0} {1} {2}", id, version, root);
-          results[id] = {version, root};
-          on_result.value()({id, version, root});
-        }
-      }
-    }
-
-    void add_result(kv::TxHistory::RequestID id, kv::Version version) override
-    {
-      auto consensus = store.get_consensus();
-
-      if (consensus != nullptr && consensus->type() == ConsensusType::BFT)
-      {
-        if (on_result.has_value())
-        {
-          auto root = get_replicated_state_root();
-          LOG_DEBUG_FMT("HISTORY: add_result {0} {1} {2}", id, version, root);
-          results[id] = {version, root};
-          on_result.value()({id, version, root});
-        }
-      }
-    }
-
-    void add_response(
-      kv::TxHistory::RequestID id,
-      const std::vector<uint8_t>& response) override
+    void append(std::shared_ptr<std::vector<uint8_t>> replicated) override
```

Comment:
Nit: No reason to copy this shared_ptr and deal with ref-count changes. Could be `const std::vector<uint8_t>&` for current needs, or at least `const std::shared_ptr...&` if we may want to extend its lifetime in future.

Reply:
Good point!